lib/pypiCoordinatesMapper.js

```diff
@@ -23,8 +23,17 @@
 
   async _resolve(coordinates) {
     const url = `${this.baseUrl}/pypi/${coordinates.name}/json`
-    const answer = await requestPromise({ url, method: 'GET', json: true })
-    return answer?.info?.name && { name: answer.info.name }
+    try {
+      const answer = await this._handleRequest(url)
+      return answer?.info?.name && { name: answer.info.name }  
+    } catch (error) {
+      if (error.statusCode === 404) return null
+      throw error
+    }
+  }
+
+  _handleRequest(url) {
+    return requestPromise({ url, method: 'GET', json: true })
```

Comment:
This private function is introduced so that the network call can easily mocked in the unit test. The host name of the url is fixed to be `https://pypi.python.org`.  The path is based on the name of the coordinates, which can potentially come from a json payload and needs to be restricted.

Reply:
@elrayle @lumaxis (PR to fix this)[https://github.com/clearlydefined/service/pull/1065] is ready for review.

Reply:
I reviewed the other PR and opened one to see if this code change makes code scanning happy 🙏🏼
https://github.com/clearlydefined/service/pull/1066